plugins/python/python.py

```diff
@@ -284,13 +284,19 @@ def _on_server_attrs_hook(word, word_eol, attrs, userdata):
 @ffi.def_extern()
 def _on_timer_hook(userdata):
     hook = ffi.from_handle(userdata)
+    plugin = hook.plugin
     if hook.callback(hook.userdata) == True:
         return 1
 
-    hook.is_unload = True  # Don't unhook
-    for h in hook.plugin.hooks:
+    try:
+        hook.is_unload = True  # Don't unhook
+    except ReferenceError:
+        # hook is a weak reference, it might have been destroyed by now
+        pass
```

Comment:
Would it work to hoist out `hook.is_unload` above `hook.callback` just like you did for plugin?

Reply:
Do keep the comment that `hook` can be destroyed in `hook.callback` though.

Reply:
Hoisting the assignment to `hook.is_unload` would prevent the finalizer `Hook.__del__` from calling `hexchat_unhook`.
Meaning that python code calling `hexchat.unhook(timer)` would remove the `Hook` from the `Plugin`, getting the `Hook` object destroyed without canceling the timer at the C level. It would get called and produce a `ReferenceError` when trying to call the registered python callback.

Setting it back to `False` if the callback returns `True` wouldn't eliminate the need for a `try` if the callback both cancels itself and return `True`.

As far as I understand, setting `hook.is_unload = True` only makes sens right before removing the hook from `plugin.hooks`. But please see my comment below as to why I think it might be ok to just remove this line.